File: Source/Domain/Library/Content/ApiContracts/ContentListItem.cs

namespace VttTools.Library.Content.ApiContracts;

public record ContentListItem {
    public Guid Id { get; init; }
    public ContentType Type { get; init; }
    public string Name { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public bool IsPublished { get; init; }
    public Guid OwnerId { get; init; }

    public AdventureStyle? Style { get; init; }
    public bool? IsOneShot { get; init; }
    public int? SceneCount { get; init; }
    public Resource? Background { get; init; }
}

File: Source/WebClientApp/src/features/content-library/types/contentListItem.ts
import type { MediaResource } from '@/types/domain';
import type { AdventureStyle } from './adventure';

export enum ContentType {
  Adventure = 0,
  Campaign = 1,
  World = 2,
}

export interface ContentListItem {
  id: string;
  type: ContentType;
  name: string;
  description: string;
  isPublished: boolean;
  ownerId: string;
  style?: AdventureStyle | null;
  isOneShot?: boolean | null;
  encounterCount?: number | null;
  background?: MediaResource | null;
}

export interface ContentItemSummary {
  id: string;
  type: ContentType;
  name: string;
  isPublished: boolean;
  thumbnailUrl?: string;
  metadata?: Record<string, unknown>;
}
